### PR TITLE
fix(Button): Add bool to propTypes of attached

### LINKED ISSUE #2104: Button: `attached` prop should accept `true` as value
### Steps

Add a `Button` with boolean `attached` prop

### Expected Result

No warnings

### Actual Result
Warning logged: 

```
"Warning: Failed prop type: Invalid prop `attached` of value `true` supplied to `t`, expected one of ['left','right','top','bottom'].
    in t (created by App)
    in App"
```

### Version
0.67.1

### Testcase
https://codepen.io/anon/pen/YrGjqx?editors=1112

### PR DESCRIPTION
**Pull request by kasbah**
fixes #2104

### CODE REVIEW

**Comment by levithomason** [comment]
Looks like we may also need to update the ButtonGroup.

**Comment by kasbah** [comment]
Looks to me like `Button.Group` doesn't attach properly in the middle currently so the warning is kind of appropriate? https://codepen.io/anon/pen/KXgrLd?editors=1111

**Comment by kasbah** [comment]
Weirdly enough, any other string will work. :confused: https://codepen.io/anon/pen/KXgbpz?editors=1111

EDIT: also some issues with the attaching of the group in general: the left side isn't rounded at all no matter what. 
image

EDIT2: nvm, [that's a known issue with single button groups](https://github.com/Semantic-Org/Semantic-UI/issues/5251)

**Comment by kasbah** [comment]
I pushed a commit that I think fixes the handling for `ButtonGroup`. Seems like [the CSS has no issues with middle attached groups](https://jsfiddle.net/u6kb1hfa/) so it should be fine, but I don't really know what I am doing, just copying and pasting things that seem to make sense and I don't yet know how to best do a manual check if this behaves as desired.

File: src/elements/Button/ButtonGroup.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   SUI,
   useKeyOnly,
   useValueAndKey,
+  useKeyOrValueAndKey,
```
Comment (layershifter) [inline src/elements/Button/ButtonGroup.js]: Please sort members in alphabetical order, i.e.:

```js
useKeyOnly,
useKeyOrValueAndKey,
useValueAndKey,
```

File: src/elements/Button/ButtonGroup.js [hunk 2]
```diff
@@ -55,8 +56,8 @@ function ButtonGroup(props) {
     useKeyOnly(secondary, 'secondary'),
     useKeyOnly(toggle, 'toggle'),
     useKeyOnly(vertical, 'vertical'),
-    useValueAndKey(attached, 'attached'),
     useValueAndKey(floated, 'floated'),
+    useKeyOrValueAndKey(attached, 'attached'),
```
Comment (layershifter) [inline src/elements/Button/ButtonGroup.js]: ^, same there

File: test/specs/elements/Button/ButtonGroup-test.js [hunk 3]
```diff
@@ -13,7 +13,8 @@ describe('ButtonGroup', () => {
     widthClass: 'buttons',
   })
 
-  common.propKeyAndValueToClassName(ButtonGroup, 'attached', ['left', 'right', 'top', 'bottom'])
+  common.propKeyOrValueAndKeyToClassName(ButtonGroup, 'attached', ['left', 'right', 'top', 'bottom'])
```
Comment (layershifter) [inline test/specs/elements/Button/ButtonGroup-test.js]: And there :smile:

**Review by layershifter — CHANGES_REQUESTED** [review]
@kasbah Thanks for PR, I've left some comments about styling. Also, `Button` should also use `useKeyOrValueAndKey` and use corresponding test.

**Comment by kasbah** [comment]
Thanks @layershifter, I fixed the code style issues. `Button` is already using `useKeyOrValueAndKey`, that's where I copied it from. ;)

**Comment by layershifter** [comment]
@kasbah Shame to me :rabbit: I've retriggered build, now it passed successfully. Great job :+1:

**Review by layershifter — APPROVED** [review]

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.74.0`

**Pull request merged**